### PR TITLE
add unit test for util

### PR DESCRIPTION
**Pull request by zcq98**
# Pull Request

- [x] Make sure you have followed [Kube-OVN Code Style](https://github.com/kubeovn/kube-ovn/blob/master/CODE_STYLE.md).

## What type of this PR

Examples of user facing changes:

- Tests

### CODE REVIEW

**Comment by zcq98** [comment]
@bobz965 Could you help review this？

File: pkg/util/net_test.go [hunk 1]
```diff
@@ -510,6 +511,69 @@ func TestIsValidIP(t *testing.T) {
...
+	}{
+		{
+			name:     "Valid IP",
+			input:    "192.168.1.1",
+			expected: nil,
+		},
+		{
+			name:     "Invalid IP",
+			input:    "invalid.ip.address",
+			expected: fmt.Errorf("node dns ip %s is not valid ip", "invalid.ip.address"),
```
Comment (bobz965) [inline pkg/util/net_test.go:528]:
```suggestion
			expected: fmt.Errorf("node dns ip %s is invalid ip", "invalid.ip.address"),
```

File: pkg/util/strings_test.go [hunk 2]
```diff
@@ -48,3 +48,31 @@ func TestDoubleQuotedFields(t *testing.T) {
...
+		input  []byte
+		output string
+	}{
+		{
+			name:   "Empty input",
+			input:  []byte(""),
+			output: "e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855",
+		},
+		{
+			name:   "Non-empty input",
```
Comment (bobz965) [inline pkg/util/strings_test.go:64]:
```suggestion
			name:   "Non empty input",
```

File: pkg/util/subnet_test.go [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+			provider: OvnProvider,
+			expected: true,
+		},
+		{
+			name:     "ovn provider with namespace",
+			provider: "namespace.cluster.ovn",
+			expected: true,
+		},
+		{
+			name:     "non-ovn provider",
```
Comment (bobz965) [inline pkg/util/subnet_test.go:29]:
```suggestion
			name:     "non ovn provider",
```

File: pkg/util/net_test.go [hunk 4]
```diff
@@ -510,6 +511,69 @@ func TestIsValidIP(t *testing.T) {
...
+			externalNet: "",
+			expected:    "ovn-vpc-external-network",
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			result := GetExternalNetwork(tt.externalNet)
+			if result != tt.expected {
+				t.Errorf("got %v, want %v", result, tt.expected)
```
Comment (bobz965) [inline pkg/util/net_test.go:571]:
```suggestion
				t.Errorf("got %v, but want %v", result, tt.expected)
```

File: pkg/util/net_test.go [hunk 5]
```diff
@@ -633,6 +697,39 @@ func TestAppendGwByCidr(t *testing.T) {
...
+			externalNets: []string{"custom-external-network1", "custom-external-network2"},
+			expected:     "custom-external-network1",
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			result := GetNatGwExternalNetwork(tt.externalNets)
+			if result != tt.expected {
+				t.Errorf("got %v, want %v", result, tt.expected)
```
Comment (bobz965) [inline pkg/util/net_test.go:727]:
```suggestion
				t.Errorf("got %v, but want %v", result, tt.expected)
```

File: pkg/util/net_test.go [hunk 6]
```diff
@@ -709,34 +806,42 @@ func TestGetIPAddrWithMask(t *testing.T) {
...
-			}
-			if c.want != ans {
-				t.Errorf("%v, %v expected %v, but %v got",
-					c.ip, c.cidr, c.want, ans)
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got, err := GetIPAddrWithMask(tt.ip, tt.cidr)
+			if (err != nil && tt.want != "") || (err == nil && got != tt.want) {
+				t.Errorf("GetIPAddrWithMask(%s, %s) = %v, %v, want %v, %v", tt.ip, tt.cidr, got, err, tt.want, nil)
```
Comment (bobz965) [inline pkg/util/net_test.go:844]:
```suggestion
				t.Errorf("got %v, %v, but want %v, %v", tt.ip, tt.cidr, got, err, tt.want, nil)
```

File: pkg/util/ovn_test.go [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+			nodeName: "this-is-a-very-long-node-name",
+			expected: "node-this-is-a-very-long-node-name",
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			result := NodeLspName(test.nodeName)
+			if result != test.expected {
+				t.Errorf("NodeLspName(%q) = %q, expected %q", test.nodeName, result, test.expected)
```
Comment (bobz965) [inline pkg/util/ovn_test.go:34]:
```suggestion
				t.Errorf("got %q, but expected %q",  result, test.expected)
```

File: pkg/util/strings_test.go [hunk 8]
```diff
@@ -48,3 +48,31 @@ func TestDoubleQuotedFields(t *testing.T) {
...
+			input:  []byte("hello"),
+			output: "2cf24dba5fb0a30e26e83b2ac5b9e29e1b161e5c1fa7425e73043362938b9824",
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got := Sha256Hash(tt.input)
+			if got != tt.output {
+				t.Errorf("Sha256Hash(%v) = %v, want %v", tt.input, got, tt.output)
```
Comment (bobz965) [inline pkg/util/strings_test.go:74]:
```suggestion
				t.Errorf("got %v, but want %v", tt.input, got, tt.output)
```

File: pkg/util/validator_test.go [hunk 9]
```diff
@@ -817,3 +817,212 @@ func TestValidateCidrConflict(t *testing.T) {
...
+			},
+			wantErr: true,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			err := ValidateVpc(tt.vpc)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("ValidateVpc() error = %v, wantErr %v", err, tt.wantErr)
```
Comment (bobz965) [inline pkg/util/validator_test.go:1024]:
```suggestion
				t.Errorf("got error = %v, but wantErr %v", err, tt.wantErr)
```

File: pkg/util/vpc_nat_gateway_test.go [hunk 10]
```diff
@@ -0,0 +1,71 @@
...
+			input:    "",
+			expected: "vpc-nat-gw-",
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			result := GenNatGwStsName(tc.input)
+			if result != tc.expected {
+				t.Errorf("Expected %s, got %s", tc.expected, result)
```
Comment (bobz965) [inline pkg/util/vpc_nat_gateway_test.go:34]:
```suggestion
				t.Errorf("Expected %s, but got %s", tc.expected, result)
```

File: pkg/util/vpc_nat_gateway_test.go [hunk 11]
```diff
@@ -0,0 +1,71 @@
...
+			input:    "",
+			expected: "vpc-nat-gw--0",
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			result := GenNatGwPodName(tc.input)
+			if result != tc.expected {
+				t.Errorf("Expected %s, got %s", tc.expected, result)
```
Comment (bobz965) [inline pkg/util/vpc_nat_gateway_test.go:67]:
```suggestion
				t.Errorf("Expected %s, but got %s", tc.expected, result)
```

**Review by bobz965 — APPROVED** [review]

**Comment by bobz965** [comment]
@zhangzujian could you please help review? thanks!

**Review by zhangzujian — APPROVED** [review]

**Pull request merged**